web/src/lib/components/StallView.svelte

```diff
@@ -1,120 +1,119 @@
+                    currentTab = "NOT RUNNING";
+                }
             }, 
-            new ErrorHandler(true, () => {
+            new ErrorHandler(false, () => {
                 loading = false;
             }));
     }
 
-    function fetchEntities() {
-        getEntities(loader, $token,
-            e => {
-                entities = e;
-            });
-    }
-
     afterNavigate(() => {
-        fetchStall(stallName);
-        fetchEntities();
-    });
-
-    onMount(async () => {
-        fetchProfile($token);
+        if (stallOwnerNym !== "") {
+            fetchStall(stallOwnerNym);
+        }
     });
 </script>
 
-
 {#if loading}
-    <div class="hero h-5/6 bg-base-200">
-        <div class="hero-content text-center">
-            <div class="max-w-md">
-                <div class="lds-ring"><div></div><div></div><div></div><div></div></div>
-            </div>
-        </div>
+    <Loading />
+{/if}
+
+{#if stallOwner && !loading}
+    <ProfileCard profileUser={stallOwner} />
+    <div class="tabs flex items-center justify-center">
+        {#each tabList as tab}
+            <li class="tab text-xs font-semibold" class:tab-active={tab === currentTab} class:tab-bordered={tab === currentTab} on:click={() => currentTab = tab}>
+                <div>{tab}</div>
+            </li>
+        {/each}
     </div>
-{:else}
-    {#if stallOwner}
-        <div class="w-full md:w-1/2 mx-auto mt-4">
-            <div class="mx-auto">
-                <!-- top profile section -->
-                <div class="flex items-center justify-between mb-4">
-                    <div class="flex-none rounded-full border border-accent avatar lg:h-32 lg:w-32 w-12 h-12 mx-4">
-                        <div class="flex justify-center">
-                            <img src={stallOwner.twitter.profileImageUrl} class="flex-none rounded-full" alt="Avatar"/>
-                        </div>
-                    </div>
-                </div>
-                <span class="font-thin text-3xl mx-4">
-                    @{stallOwner.nym}
-                </span>
-                {#if entities === null || entities.length === 0}
-                    <div class="mt-4 mx-4 font-thin text-xl mb-4">
-                        User is not selling anything
-                    </div>
+    <div class="pt-6 pb-6">
+        {#if currentTab === 'NOT RUNNING'}
+            <div class="h-auto">
+                <ListView
+                    title="{stallOwner.nym}'s Stall"
+                    loader={{endpoint: `users/${stallOwner.nym}/auctions?filter=new`, responseField: 'auctions', fromJson}}
+                    newEntity={() => new Auction()}
+                    editor={AuctionEditor}
+                    showNewButton={true}
+                    card={AuctionCard}
+                    showAsGrid={true} />
+            </div>
+        {:else if currentTab === 'ACTIVE AUCTIONS'}
+            <div class="h-auto">
+                {#if card === "AuctionCard"}
```

Comment:
Why do we need the if/else? You can just have card variable be the type itself rather than a string, so you could just pass it further to ListView, no?